test/functional/pt_osc_migration_functional_test.rb

```diff
@@ -38,6 +38,22 @@ class TestMigration < ActiveRecord::PtOscMigration; end
           ActiveRecord::Base.connection.execute "DROP TABLE IF EXISTS `#{@table_name}`;"
         end
 
+        context 'a raw SQL execute migration' do
+          setup do
+            TestMigration.class_eval <<-EVAL
+            def up
+              execute 'SELECT 1'
+            end
+            EVAL
+          end
+
+          should 'throw an exception' do
+            assert_raise ActiveRecord::UnsupportedMigrationError do
+              @migration.migrate(:up)
+            end
+          end
```

Comment:
The problem this is preventing isn't `pt-online-schema-change` with bad data, it's ActiveRecord/Rails migration executing the DDL synchronously _not_ through `pt-online-schema-change`. A non-functioning migration would have been great (in comparison), and I might not have bothered with this PR!